tck/src/main/java/org/reactivestreams/tck/TestEnvironment.java

```diff
@@ -50,15 +50,47 @@
    * the implementation, but can in some cases result in longer time to
    * run the tests.
    * @param defaultTimeoutMillis default timeout to be used in all expect* methods
+   * @param defaultPollTimeoutMillis default amount of time to poll for events if {@code defaultTimeoutMillis} isn't
+   *                                 preempted by an asynchronous event.
    * @param defaultNoSignalsTimeoutMillis default timeout to be used when no further signals are expected anymore
    * @param printlnDebug         if true, signals such as OnNext / Request / OnComplete etc will be printed to standard output,
    */
-  public TestEnvironment(long defaultTimeoutMillis, long defaultNoSignalsTimeoutMillis, boolean printlnDebug) {
+  public TestEnvironment(long defaultTimeoutMillis, long defaultNoSignalsTimeoutMillis, long defaultPollTimeoutMillis,
+                         boolean printlnDebug) {
     this.defaultTimeoutMillis = defaultTimeoutMillis;
+    this.defaultPollTimeoutMillis = defaultPollTimeoutMillis;
```

Comment:
I think we should require (`if ... else throw`) that `defaultPollTimeoutMillis` be equal or smaller than `defaultTimeoutMillis`, otherwise we could get into some weird situations.

Reply:
I can do this if you prefer, however we need to take the `min` between "remaining total time" and "poll time" anyways so this restriction is not necessary. For consistency we would also have to enforce it in `expectError` which may pop-up in unexpected cases.

Reply:
Oh true, the `min` handles it there quite nicely. Seems like a good idea to rely on that and not push more throws in there.

It could cause an issue if someone used the `expectError` with passed in timeout values explicitly -- noone really should have, but can't know if they did 🤔 

Sounds good to keep as is 👍